### PR TITLE
Fisrt version

### PR DESCRIPTION
**Pull request by trocolit2**
It is the first version of image stitching orogen.
This component is able to run, but not in real time (more than 5 fps).
Please review this code @gustavoneves12, @romulogcerqueira or @dbcesar .

### CODE REVIEW

**Comment by trocolit2** [comment]
[Photo Mosaic: Image Stitching orogen component](https://trello.com/c/gythJU89/27-photo-mosaic-image-stitching-orogen-component)

**Comment by trocolit2** [comment]
Applied suggested modifications

**Comment by trocolit2** [comment]
@romulogcerqueira or @gustavoneves12 please review!

File: tasks/Task.cpp [hunk 1]
```diff
@@ -28,17 +74,56 @@ bool Task::configureHook()
...
+    method_ = _image_stitching.value();
+
     return true;
 }
 bool Task::startHook()
 {
     if (! TaskBase::startHook())
         return false;
+
+    image_stitching_ = new ImageStitching();
```
Comment (gustavoneves12) [inline tasks/Task.cpp]: It would be good you initialize the **image_stitching_** as null in the constructor.

File: tasks/Task.cpp [hunk 2]
```diff
@@ -16,10 +62,10 @@ Task::Task(std::string const& name, RTT::ExecutionEngine* engine)
 
 Task::~Task()
 {
+    delete image_stitching_;
```
Comment (gustavoneves12) [inline tasks/Task.cpp]: It is a good practice check if the pointer is null before delete. See below:

```
if (image_stitching_) {
    delete image_stitching_;
    image_stitching_ = NULL;
}
```

File: tasks/Task.cpp [hunk 3]
```diff
@@ -4,6 +4,52 @@
...
+void SelectImageStitchingMethod( ImageStitching *image_stitching,
+                                 ImageStitchingMethods method){
+
+    switch (method) {
+        case OPENCV_STITCHING:
+        {
+            OpenCVImageStitchingStruct conf;
+            conf.type_warper = PLANE_WARPER;
+            OpenCVImageStitching *opencv_image_stiching;
+            opencv_image_stiching = new OpenCVImageStitching(conf);
```
Comment (gustavoneves12) [inline tasks/Task.cpp]: Where are you deleting this one? You can change this for:

```
ImageStitchingMethod* CreateImageStitchingMethod(method) {
      . . . 
      return new OpenCVImageStitching(conf);
}
```

File: tasks/Task.hpp [hunk 4]
```diff
@@ -4,50 +4,71 @@
...
+class Task : public TaskBase {
+    friend class TaskBase;
+
+    private:
 
+        int images_counter_;
+        std::vector<cv::Mat> vector_images_;
+
+        ImageStitchingMethods method_;
+        ImageStitching *image_stitching_;
```
Comment (gustavoneves12) [inline tasks/Task.hpp]: Does the **image_stitching_**  need to be a pointer? IMO, you can use a static object here.

**Comment by trocolit2** [comment]
[Photo Mosaic: Image Stitching orogen component](https://trello.com/c/gythJU89/27-photo-mosaic-image-stitching-orogen-component)

**Comment by trocolit2** [comment]
@gustavoneves12 and @romulogcerqueira, can you review?

**Pull request merged**